language/stdlib/modules/LibraAccount.move

```diff
@@ -836,15 +849,17 @@ module LibraAccount {
         gas_units_remaining: u64
     ) acquires LibraAccount, Balance, AccountOperationsCapability {
         let sender = Signer::address_of(account);
-        // Load the transaction sender's account and balance resources
-        let sender_balance = borrow_global_mut<Balance<Token>>(sender);
 
         // Charge for gas
         let transaction_fee_amount = txn_gas_price * (txn_max_gas_units - gas_units_remaining);
-        assert(
-            balance_for(sender_balance) >= transaction_fee_amount,
-            6
-        );
+        if (transaction_fee_amount > 0)  {
+            // Load the transaction sender's balance resource only if the txn fee is non-zero
+            let sender_balance = borrow_global_mut<Balance<Token>>(sender);
```

Comment:
This is currently guaranteed to succeed because of the check in the prologue (right)?

If we ever add functionality for removing a `Balance` resource with value 0 (which seems reasonable enough), I think this will no longer be guaranteed to succeed. It would be nice to rewrite this in a way that ensures the epilogue will fail with 6 even if someone:
- Sends a tx with a nonzero fee in LBR
- Transfers all of their LBR elsewhere
- Deletes their (now-zero) `Balance<LBR>` resource

E.g.,

```
let sender_balance = if (exists<Balance<Token>>(sender)) {
    balance(sender)
} else {
    0
};
assert(balance_for(sender_balance) >= transaction_fee_amount, 6);
```

Reply:
ahhh good point! I'll make that update :) (and check with ops/AOS to make sure it looks good with them).